docs/TuxReact.md

```diff
@@ -0,0 +1,449 @@
+            <ol>
+              <li><a href="#React-Components-props-children">props.children</a></li>
+              <li><a href="#React-Components-PropTypes">PropTypes</a></li>
+            </ol>
+          </li>
+          <li><a href="#React-Components-Lifecycle-Methods">Lifecycle Methods</a></li>
+        </ol>
+      </li>
+    </ol>
+  </li>
+  <li><a href="#TuxedoJS-and-React">TuxedoJS and React</a>
+    <ol>
+      <li><a href="#TuxedoJS-OwnerClass">TuxedoJS OwnerClass</a></li>
+      <li><a href="#TuxedoJS-OwneeClass">TuxedoJS OwneeClass</a></li>
+      <li><a href="#TuxedoJS-MutableClass">TuxedoJS MutableClass</a></li>
+    </ol>
+  </li>
+</ol>
+
+## <a id="#Premise"></a>Premise [#](#Premise)
+>`React` is an extremely powerful and complex view library. `TuxedoJS` builds on its functionality in order to make it more semantic and easier to work with.
+
+***
+
+### <a id="React-Components-in-a-Nutshell"></a>React Components in a Nutshell [#](#React-Components-in-a-Nutshell)
+React is a view library designed for extremely high performance rendering. The core concept in `React` is the Virtual DOM. When `React` components update they create a virtual version of the new DOM. Then, `React` diffs the new virtual DOM with the old copy and submits only those changes between the two in order to update the real DOM. `React` is a complicated library and we recommend reviewing the full docs [here](http://facebook.github.io/react/). However, we figured it would be valuable tp provide a quick primer on `React` to give an introductory understanding.
+
+***
+
+#### <a id="Writing-React-and-tuxReact-in-JSX"></a>1) Writing React and tux/React in JSX [#](#Writing-React-and-tuxReact-in-JSX)
+`React` can be written beautiful [JSX](http://facebook.github.io/react/docs/displaying-data.html) which allows you to write React components as though they were HTML tags. We love JSX here at TuxedoJS and we stongly recommend using it. Our components are written using JSX but you do not need to write JSX with Tux. Our `React` examples in our docs will be written in JSX syntax.
+
+One exremely important note, if you want to use JSX with Tux (and you really should) you will need to call your `tux/React` variable `React`:
+
+```
+    var React = require('tux/React');
+```
+
+Why do we need to do this? Because JSX translates into `React.METHOD NAME` automatically, thus we need to call our variable `React` for our methods to work after compiling our JSX.
+
+In the following examples we will be writing in basic `React` syntax.
+
+***
+
+#### <a id="Rendering-in-React"></a>2) Rendering in React [#](#Rendering-in-React)
+The most common way to render a component in `React` is with the `React.render` method:
+
+```
+    var React = require('react');
+    var MyComponent = require('./components/MyComponent'); //Some component we want to render
+    React.render(<MyComponent />, document.body); //Using JSX for MyComponent
```

Comment:
Recommend explaining that in this line that the first argument is the compnent you are rendering and the second argument is where you are rendering it.